src/main/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/producer/OutgoingMessageProducer.java

```diff
@@ -0,0 +1,68 @@
+package uk.gov.companieshouse.chipsrestinterfacesconsumer.producer;
+
+import org.apache.avro.Schema;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.chipsrestinterfacesconsumer.common.ApplicationLogger;
+import uk.gov.companieshouse.chipsrestinterfacesconsumer.model.ChipsKafkaMessage;
+import uk.gov.companieshouse.kafka.message.Message;
+import uk.gov.companieshouse.kafka.producer.CHKafkaProducer;
+import uk.gov.companieshouse.service.ServiceException;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+@Component
+public class OutgoingMessageProducer implements MessageProducer {
+
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd MMM yyyy HH:mm:ss");
+
+    private ApplicationLogger logger;
+
+    private AvroSerializer avroSerializer;
+
+    private CHKafkaProducer producer;
+
+    @Autowired
+    @Qualifier("chips-kafka-send")
+    private Schema schema;
+
+    @Value("${kafka.retry.topic}")
+    private String retryTopicName;
+
+    @Autowired
+    public OutgoingMessageProducer(ApplicationLogger logger,
+                                   AvroSerializer avroSerializer,
+                                   CHKafkaProducer producer){
+        this.logger = logger;
+        this.avroSerializer = avroSerializer;
+        this.producer = producer;
+    }
+
+
+    @Override
+    public void writeToQueue(ChipsKafkaMessage chipsMessage) throws ServiceException {
+        try {
+            Message kafkaMessage = new Message();
```

Comment:
add some logging to indicate message is to be written to topic

Reply:
logging added.